include/swift/AST/SourceFile.h

```diff
@@ -344,6 +344,11 @@ class SourceFile final : public FileUnit {
     return BufferID;
   }
 
+  void setBufferID(int Val) {
```

Comment:
Why does the `SourceFile` buffer ID need to be updated?  What uses this info during completion?

Reply:
At least `ParseMembersRequest::evaluate()` uses `getBufferID()` for delayed member list parsing. https://github.com/apple/swift/blob/53c008f32cf7e83022c9f6643c960015b7f91da7/lib/Parse/ParseRequests.cpp#L36-L44

We can modify this to get bufferID from the decl location, like:
```cpp
  auto &SM = idc->getASTContext().SourceMgr;
  unsigned bufferID = SM.findBufferContainingLoc(idc->getDecl()->getLoc());
```
But I'm not sure this is the only usage of `getBufferID()` during completion.
What is your concern about setting the buffer id?

Reply:
I guess that's fine, it's just more state that we're mutating so I wanted to understand what effect it has.